src/constants/mapboxStreetsV8Fields.ts

```diff
@@ -135,42 +31,227 @@ export const STREETS_V8_FIELDS = {
+        'american_football',
+        'athletics',
+        'caravan_site',
+        'rock',
+        'muslim',
+        'skateboard',
+        'wetland',
+        'bowls',
+        'picnic_site',
+        'boules',
+        'cricket',
+        'dog_park',
+        'running',
+        'conservation',
+        'track',
+        'netball',
+        'underground',
+        'lane',
+        'rugby_union',
+        'zoo',
+        'hockey',
+        'shooting',
+        'downhill',
+        'jewish',
+        'field',
+        'football',
+        'table_tennis',
+        'handball',
+        'rough',
+        'field_hockey',
+        'team_handball',
+        'carports',
+        'pelota',
+        'rugby',
+        'paddle_tennis',
+        'archery',
+        'horse_racing',
+        'gaelic_games',
+        'softball',
+        'golf',
+        'ice_hockey',
+        'basin',
+        'coastline',
+        'badminton',
+        'driving_range',
+        'bog',
+        'cricket_nets',
+        'swimming',
+        'futsal'
+      ] as const
     }
   },
 
-  water: {
-    // Water has no filterable fields
-  },
-
+  // ============ waterway ============
   waterway: {
+    iso_3166_1: {
+      description: 'iso_3166_1 field',
+      values: [
+        'EG',
+        'ET',
+        'CD',
+        'ZA',
+        'TZ',
+        'KE',
+        'SD',
+        'UG',
+        'MA',
+        'DZ',
+        'GH',
+        'CI',
+        'CM',
+        'MG',
+        'MZ',
+        'NG',
+        'NE',
+        'BF',
+        'MW',
+        'ML',
+        'TD',
+        'SN',
+        'AO',
+        'ZW',
+        'CN',
+        'IN',
+        'ID',
+        'PK',
+        'BD',
+        'RU',
+        'JP',
+        'PH',
+        'VN',
+        'TR',
+        'IR',
+        'TH',
+        'MM',
+        'KR',
+        'IQ',
+        'AF',
+        'MY',
+        'NP',
+        'DE',
+        'FR',
+        'GB',
+        'IT',
+        'ES',
+        'UA',
+        'PL',
+        'RO',
+        'NL',
+        'GR',
+        'HR',
+        'BE',
+        'PT',
+        'CZ',
+        'HU',
+        'BY',
+        'SE',
+        'AT',
+        'CH',
+        'BG',
+        'RS',
+        'DK',
+        'FI',
+        'US',
+        'MX',
+        'CA',
+        'GT',
+        'CU',
+        'HT',
+        'DO',
+        'HN',
+        'NI',
+        'SV',
+        'CR',
+        'PR',
+        'PA',
+        'JM',
+        'TT',
+        'GP',
+        'MQ',
+        'AU',
+        'PG',
+        'NZ',
+        'FJ',
+        'MU',
+        'RE',
+        'MV',
+        'SC',
+        'BR',
+        'CO',
+        'AR',
+        'PE',
+        'VE',
+        'CL',
+        'EC',
+        'BO',
+        'PY',
+        'UY'
+      ] as const
+    },
+    iso_3166_2: {
+      description: 'iso_3166_2 field',
+      values: [] as const // string
+    },
     class: {
-      description: 'Waterway classification',
+      description: 'class field',
```

Comment:
[nit] if those `description` are only `<key> field`, it will not help with the context, consider to remove them.